ui-src/src/app/configurations/configurations.component.ts

```diff
@@ -4,139 +4,36 @@ import { AutomatedUIServices } from "../automated-ui-services";
 })
 export class ConfigurationsComponent implements OnInit {
   formData = {};
+  // needed for layout to load
+  configSchema = { layout: "" };
+
   constructor(
     private automatedUIServices: AutomatedUIServices,
-    private toastr: ToastsManager
+    private toastr: ToastsManager,
+    private http: Http
   ) {}
 
-  configSchema = {
-    type: "object",
-    properties: {
-      apiName: { type: "string" },
-      targetHost: { type: "string" },
-      targetPort: {
-        type: "string",
-        minimum: 1,
-        maximum: 65535
-      },
-      memoryEndpoint: { type: "string" },
-      numIterations: { type: "integer", minimum: 1 },
-      concurrentUsers: { type: "integer", minimum: 1 },
-      allowablePeakMemoryVariance: {
-        type: "number",
-        minimum: 0,
-        maximum: 100
-      },
-      allowableServiceResponseTimeVariance: {
-        type: "number",
-        minimum: 0,
-        maximum: 100
-      },
-      testSuite: {
-        type: "string",
-        enum: ["Default-1", "Default-2", "Default-3"]
-      },
-      requestDelay: { type: "integer", minimum: 1 },
-      TPSFreq: { type: "integer", minimum: 1 },
-      rampUsers: { type: "integer", minimum: 1 },
-      rampDelay: { type: "integer", minimum: 0 },
-      testCaseDir: { type: "string" },
-      testSuiteDir: { type: "string" },
-      baseStatsOutputDir: { type: "string" },
-      reportOutputDir: { type: "string" }
-    },
-    required: [
-      "apiName",
-      "targetHost",
-      "targetPort",
-      "numIterations",
-      "concurrentUsers",
-      "allowablePeakMemoryVariance",
-      "allowableServiceResponseTimeVariance",
-      "testSuite",
-      "requestDelay",
-      "TPSFreq",
-      "rampUsers",
-      "rampDelay",
-      "testCaseDir",
-      "testSuiteDir",
-      "baseStatsOutputDir",
-      "reportOutputDir"
-    ]
-  };
-
-  exampleData = {
-    allowablePeakMemoryVariance: 15,
-    allowableServiceResponseTimeVariance: 15
-  };
-
-  layout = [
-    {
-      type: "flex",
-      "flex-flow": "row wrap",
-      items: [
-        "apiName",
-        "numIterations",
-        {
-          key: "requestDelay",
-          title: "Request Delay (ms)"
-        }
-      ]
-    },
-    {
-      type: "flex",
-      "flex-flow": "row wrap",
-      items: [
-        "targetHost",
-        "concurrentUsers",
-        {
-          key: "TPSFreq",
-          title: "TPS Frequency (s)"
-        }
-      ]
-    },
-    {
-      type: "flex",
-      "flex-flow": "row wrap",
-      items: [
-        "targetPort",
-        {
-          key: "allowablePeakMemoryVariance",
-          title: "Memory Variance (%)"
-        },
-        "rampUsers"
-      ]
-    },
-    {
-      type: "flex",
-      "flex-flow": "row wrap",
-      items: [
-        "memoryEndpoint",
-        {
-          key: "allowableServiceResponseTimeVariance",
-          title: "Service Variance (%)"
+  ngOnInit() {
+    this.http
+      .get("assets/schema.json")
```

Comment:
This should be in the configuration.service.js - (aka the automatedUIServices). If your adding this in there can you rename the automatedUIServices to configuration.service.js

the schema json should not be part of the front end assets, it should come from the api services. As if the file is changed, it is changed in one place and there should no need to make any changes on the front end. At the moment if you make a change to the schema on the services side you will also need to make the same change on the front end.